### PR TITLE
refactor: use deno_ast

### PR DESCRIPTION
**Pull request by dsherret**
This change makes deno_lint depend on deno_ast.

Pending changes in deno_ast PR: denoland/deno_ast#1

cc @magurotuna

### CODE REVIEW

File: src/ast_parser.rs [hunk 1]
```diff
@@ -71,19 +59,12 @@ impl fmt::Display for SwcDiagnostic {
 }
 
 impl SwcDiagnostic {
```
Comment (dsherret) [inline src/ast_parser.rs:61]: We could maybe remove this and just use the `deno_ast` diagnostic?

File: src/ast_parser.rs [hunk 2]
```diff
@@ -117,47 +98,27 @@ impl AstParser {
...
+        source: deno_ast::ParsedSourceTextInfo::from_string(
+          source_code.to_string(),
+        ),
+        capture_tokens: true,
+        maybe_syntax: Some(syntax),
+      },
+      |program| {
+        // This is used to apply proper "syntax context" to all AST elements. When SWC performs
+        // transforms/folding it might change some of those context and "ts_resolver" ensures
+        // that all elements end up in proper lexical scope.
```
Comment (dsherret) [inline src/ast_parser.rs:115]: Copied and pasted this from Bartek's comment.

File: src/ast_parser.rs (hunk 1 above)
Comment (bartlomieju) [reply src/ast_parser.rs:61]: Sounds good to me, feel free to do that in a follow up

**Review by bartlomieju — APPROVED** [review]
LGTM, 130 net lines removed, can't ask for more :)

**Review by magurotuna — APPROVED** [review]
LGTM 🙆

**Comment by magurotuna** [comment]
@dsherret can you rebase with main? https://github.com/denoland/deno_lint/pull/836 may cause CI on this PR to fail.

**Comment by dsherret** [comment]
@magurotuna yup, I’ll rebase before merging. I’ll probably do it on Monday.

**Comment by dsherret** [comment]
Will do a follow up PR with some changes. Just want to get this merged for now.

**Pull request merged**